install/kubernetes/cilium/templates/cilium-operator/deployment.yaml

```diff
@@ -135,14 +135,18 @@ spec:
         - name: AZURE_RESOURCE_GROUP
           value: {{ .Values.azure.resourceGroup }}
         {{- end }}
-        {{- if .Values.azure.clientID }}
         - name: AZURE_CLIENT_ID
-          value: {{ .Values.azure.clientID }}
-        {{- end }}
-        {{- if .Values.azure.clientSecret }}
+          valueFrom:
+            secretKeyRef:
+              name: cilium-azure
+              key: AZURE_CLIENT_ID
+              optional: true
```

Comment:
I would advocate against using the `optional` flag since both secrets are required for the operator to actually work. What do you think?

Reply:
I agree - only put it there to replicate the existing if statements but I will happily remove it.
The only reason for keeping it would be for future [aad-pod-identity](https://github.com/Azure/aad-pod-identity) support.